### PR TITLE
feat: allow replacement transaction verification after prebuild

### PR DESCRIPTION
**Pull request by rushilbg**
[feat: allow replacement transaction verification after prebuild](https://github.com/BitGo/BitGoJS/pull/4218/commits/6a26b176b2dff889c757acaf6728864254d678cb) 

The verification will happen based on the recipients of the original
transaction that is being replaced instead of the recipients of the
replacement transaction API request, which is empty always.

[BTC-820](https://bitgoinc.atlassian.net/browse/BTC-820)

[BTC-820]: https://bitgoinc.atlassian.net/browse/BTC-820?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/abstract-utxo/src/abstractUtxoCoin.ts [hunk 1]
```diff
@@ -519,12 +519,24 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
 
     const allOutputs = [...explanation.outputs, ...explanation.changeOutputs];
 
-    // verify that each recipient from txParams has their own output
-    const expectedOutputs = _.get(txParams, 'recipients', [] as TransactionRecipient[]).map((output) => {
-      return { ...output, address: this.canonicalAddress(output.address) };
-    });
+    let expectedOutputs;
+    if (txParams.rbfTxIds) {
```
Comment (OttoAllmendinger) [inline modules/abstract-utxo/src/abstractUtxoCoin.ts:523]:
```suggestion
    if (txParams.rbfTxIds) {
      assert(txParams.rbfTxIds.length === 1);
```

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
nit

File: modules/abstract-utxo/src/abstractUtxoCoin.ts [hunk 2]
```diff
@@ -158,6 +158,7 @@ export interface TransactionPrebuild<TNumber extends number | bigint = number> e
 export interface TransactionParams extends BaseTransactionParams {
   walletPassphrase?: string;
   changeAddress?: string;
+  rbfTxIds?: string;
```
Comment (OttoAllmendinger) [inline modules/abstract-utxo/src/abstractUtxoCoin.ts:161]:
```suggestion
  rbfTxIds?: string[];
```
?

**Review by OttoAllmendinger — COMMENTED** [review]
maybe I am missing something

**Review by OttoAllmendinger — DISMISSED** [review]

**Review by OttoAllmendinger — APPROVED** [review]

**Review by ayush-9 — APPROVED** [review]

**Pull request merged**